sdcm/nemesis.py

```diff
@@ -445,18 +445,79 @@ def disrupt_nodetool_cleanup(self):
+        self._modify_table_property(name="gc_grace_seconds", val=random.randint(216000, 864000))
+
+    def disrupt_modify_table_caching(self):
+        prop_val = dict(
+            keys=random.choice(["NONE", "ALL"]),
+            rows_per_partition=random.choice(["NONE", "ALL", random.randint(1, 10000)])
+        )
+        self._modify_table_property(name="caching", val=str(prop_val))
+
+    def disrupt_modify_table_bloom_filter_fp_chance(self):
+        self._modify_table_property(name="bloom_filter_fp_chance", val=random.random()/2)
+
+    def disrupt_modify_table_compaction(self):
+        # TODO: Sub-properties for each of compaction strategies should also be tested
+        strategies = ("SizeTieredCompactionStrategy", "DateTieredCompactionStrategy",
+                      "TimeWindowCompactionStrategy", "LeveledCompactionStrategy")
+        prop_val = {"class": random.choice(strategies)}
+        self._modify_table_property(name="compaction", val=str(prop_val))
+
+    def disrupt_modify_table_compression(self):
+        algos = ("",  # no compression
+                 "LZ4Compressor",
+                 "SnappyCompressor",
+                 "DeflateCompressor")
+        algo = random.choice(algos)
+        prop_val = {"sstable_compression": algo}
+        if algo:
+            prop_val["chunk_length_kb"] = random.choice(["4K", "64KB", "1M"])
+            prop_val["crc_check_chance"] = random.random()
+        self._modify_table_property(name="compression", val=str(prop_val))
+
+    def disrupt_modify_table_crc_check_chance(self):
+        self._modify_table_property(name="crc_check_chance", val=random.random())
+
+    def disrupt_modify_table_dclocal_read_repair_chance(self):
+        self._modify_table_property(name="dclocal_read_repair_chance", val=random.random())
+
+    def disrupt_modify_table_default_time_to_live(self):
+        self._modify_table_property(name="default_time_to_live", val=630720000)  # max allowed TTL - 20 years
+
+    def disrupt_modify_table_max_index_interval(self):
+        self._modify_table_property(name="max_index_interval", val=random.choice([1024, 4096, 8192]))
+
+    def disrupt_modify_table_min_index_interval(self):
+        self._modify_table_property(name="min_index_interval", val=random.choice([128, 256, 512]))
+
+    def disrupt_modify_table_memtable_flush_period_in_ms(self):
+        self._modify_table_property(name="memtable_flush_period_in_ms", val=random.randint(1, 5000))
+
+    def disrupt_modify_table_read_repair_chance(self):
+        self._modify_table_property(name="read_repair_chance", val=random.random())
+
+    def disrupt_modify_table_speculative_retry(self):
+        options = ("'ALWAYS'",
+                   "'%spercentile'" % random.randint(1, 99),
+                   "'%sms'" % random.randint(1, 1000))
+        self._modify_table_property(name="speculative_retry", val=random.choice(options))
+
 
```

Comment:
@bentsi, despite what we said in the beginning... I think we need the wrap all these new nemesis as ONE nemesis that will randomize across all of them.
We don't really know how all these schema changes will affect our testing and too many schema changes one after another can make the issues impossible to investigate.

Reply:
OK, I will wrap this in on one disrupt method